ext/symengine/ruby_ntheory.c

```diff
@@ -0,0 +1,70 @@
+#include "ruby_ntheory.h"
+
+VALUE cntheory_nextprime(VALUE self, VALUE operand1) {
+    return function_onearg(ntheory_nextprime, operand1);
+}
+
+VALUE cntheory_gcd(VALUE self, VALUE operand1, VALUE operand2) {
+    return function_twoarg(ntheory_gcd, operand1, operand2);
+}
+
+VALUE cntheory_lcm(VALUE self, VALUE operand1, VALUE operand2) {
+    return function_twoarg(ntheory_lcm, operand1, operand2);
+}
+
+VALUE cntheory_mod(VALUE self, VALUE operand1, VALUE operand2) {
+    VALUE ans = function_twoarg(ntheory_mod, operand1, operand2);
+    VALUE ans1 = cbasic_add(ans, operand2);
+    return function_twoarg(ntheory_mod, ans1, operand2);
+}
+
+VALUE cntheory_quotient(VALUE self, VALUE operand1, VALUE operand2) {
+    return function_twoarg(ntheory_quotient, operand1, operand2);
+}
+
+VALUE cntheory_fibonacci(VALUE self, VALUE operand1){
+    basic_struct *cresult;
+    VALUE result;
+
+    unsigned long cbasic_operand1;
+    cbasic_operand1 = NUM2ULONG(operand1);
+
+    cresult = basic_new_heap();
+    ntheory_fibonacci(cresult, cbasic_operand1);
+    result = Data_Wrap_Struct(Klass_of_Basic(cresult), NULL , cbasic_free_heap, cresult);
+
+    return result;
+}
+
+VALUE cntheory_lucas(VALUE self, VALUE operand1){
+    basic_struct *cresult;
+    VALUE result;
+
+    unsigned long cbasic_operand1;
+    cbasic_operand1 = NUM2ULONG(operand1);
```

Comment:
now, it gives an error `TypeError: no implicit conversion of SymEngine::Integer into Integer`

Can I use `sympify` for this as well? In C++, the method signature is `SymEngine::lucas(unsigned long n)`. So the cwrapper also expects a unsinged long.